### ISSUE TITLE
Block the none driver from being used on non-Linux platforms

### ISSUE THREAD

**Issue by newsoulbr** [kubernetes/minikube#4498]
**The exact command to reproduce the issue**:
minikube start --vm-driver=none

**The full output of the command that failed**:
minikube v1.1.1 on darwin (amd64)
💡  Tip: Use 'minikube start -p <name>' to create a new cluster, or 'minikube delete' to delete this one.
🔄  Restarting existing none VM for "minikube" ...
⌛  Waiting for SSH access ...

💣  Failed to enable container runtime: running command: sudo systemctl start docker: exit status 1

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new

**The operating system version**:
macOS Mojave 10.14.4

**Suggestion**:
In the macOS another option is to install docker using a dmg. For the ones that use it, the driver could just identify the process and use it by the api

**Comment by afbjorklund** [comment]
Using the `none` driver means that the programs are running on the localhost.
It is not possible to run the `kubelet` and others natively on a mac, so it won't work.

What are you looking for is another minikube feature called "VM-less", see #4389 
Note that it will _still_ run within a VM, except that it will be in Docker's VM instead.

**Comment by afbjorklund** [comment]
We should probably try to give a better error message for this scenario.

**Comment by medyagh** [comment]
we are working on adding support for vm free minikube on all platfroms, please stay tuned 
https://github.com/kubernetes/minikube/issues/4389

**Comment by tstromberg** [comment]
This was fixed in v1.4

```
😄  minikube v1.4.0 on Darwin 10.14.6
💣  The driver 'none' is not supported on darwin
```

**Comment by ziaagikian** [comment]
Appeared again in v1.5.x 

```
$ minikube start --vm-driver=none
😄  minikube v1.5.2 on Darwin 10.15
💣  The driver 'none' is not supported on darwin
```
